### DISCUSSION TITLE
AWS Distributed Deployment... security group?

### DISCUSSION THREAD

**Discussion by mtoback** [Security-Onion-Solutions/securityonion#9084] [Unsupported Versions] [answered]
I created a distributed AWS from the latest AMI (mgr, search and sensor, using defaults) and I can see everything except data. Kibana shows zilch, as does hunt, alerts, etc. I see the nodes running in grafana and so-status seem to be happy on all nodes.

So I suspect that I have a security group issue.  my filebeat log shows this on the sensor instance: Filebeat is unable to load the ingest pipelines for the configured modules because the Elasticsearch output is not configured/enabled

Filebeat on manager says this over and over: 2022-11-07T23:37:13.839Z	ERROR	[logstash]	logstash/async.go:280	Failed to publish events caused by: EOF
2022-11-07T23:37:13.839Z	ERROR	[logstash]	logstash/async.go:280	Failed to publish events caused by: client is not connected
2022-11-07T23:37:15.653Z	ERROR	[publisher_pipeline_output]	pipeline/output.go:180	failed to publish events: client is not connected

search filebeat says this: 2022-11-07T17:15:21.606Z	INFO	[esclientleg]	eslegclient/connection.go:284	Attempting to connect to Elasticsearch version 7.17.1
2022-11-07T17:15:21.678Z	INFO	[modules]	fileset/pipelines.go:133	Elasticsearch pipeline loaded.	{"pipeline": "filebeat-7.17.1-zscaler-zia-pipeline"}

I suspect that it is a security group somewhere? I have rules open to 22, 80, 443, 1514, 1515, 3142, 4505, 4506, 5000, 5644, 8086, 8080, 8086  to the entire subnet and can lock it down later. I suspect I am missing something

**Comment by weslambert** [comment, marked as answer]
What does the status for each node look like in the Grid section of SOC?

If you run the following command on the manager, is the count greater than zero?

`sudo so-redis-count`

What do you see if you run the following command on a search node?

`sudo so-elasticsearch-indices-list`

Are you able to connect to port 5644 on the manager from the sensor or the search node, using something like netcat (`nc`)?

Ex. `nc -vz $manager_ip 5644`

Are the sensor and/or search node able to resolve the hostname of the manager?

Are there any clues in any of the nodes' setup log (`/root/sosetup.log`)?

**Reply by mtoback** [reply]
all three nodes are marked OK in the grid

sudo so-redis-count: 598716 and climbing

sudo so-elasticsearch-indices-list
health status index            uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   .geoip_databases IGCSHy6BR4WitB5RALWzJw   1   0         41           21     39.3mb         39.3mb
green  open   .security-7      xQvkO4IwSAaoO27x1rs9aw   1   0          4            0     30.2kb         30.2kb


nc -vz 10.7.50.168 5644
Ncat: Version 7.50 ( https://nmap.org/ncat )
Ncat: Connected to 10.7.50.168:5644.
Ncat: 0 bytes sent, 0 bytes received in 0.01 seconds.

both sensor and search are able to resolve the hostname of the manager

search/sosetup.log: I see this:
Waiting for ElasticSearch..........................connected!
   3317                   Loading templates...
   3318                   {"error":{"root_cause":[{"type":"status_exception","reason":"Cluster state has not been recovered yet, cannot write to the [null] index"}],"type":"status_ex        ception","reason":"Cluster state has not been recovered yet, cannot write to the [null] index"},"status":503}{"error":{"root_cause":[{"type":"status_exception","reason":"Clus        ter state has not been recovered yet, cannot write to the [null] index"}],"type":"status_exception","reason":"Cluster state has not been recovered yet, cannot write to the [n        ull] index"},"status":503}{"error":{"root_cause":[{"type":"status_exception","reason":"Cluster state has not been recovered yet, cannot write to the [null] index"}],"type":"s        tatus_exception","reason":"Cluster state has not been recovered yet, cannot write to the [null] index"},"status":503}{"error":{"root_cause":[{"type":"status_exception","reaso        n":"Cluster state has not been recovered yet, cannot write to the [null] index"}],"type":"status_exception","reason":"Cluster state has not been recovered yet, cannot write t        o the [null] index"},"status":503}

no errors on sensor or manager seen

**Reply by mtoback** [reply]
Also Kibana shows nothing and has the following error through a popup:  
Error fetching fields for index pattern *:so-*  (ID: ....) 
[object Object]
Reason: no_such_remote_cluster_exception. [onion-search1_searchnode]
_request/<https://10.7.50.168/kibana/.../dataViews.js:1:37078

**Reply by weslambert** [reply]
Have you checked the firewall rules/security group to ensure that communication to Redis is allowed from the search node(s) to the manager on port 9696?

Similarly, is port 9300 (Elasticsearch cluster communication) allowed between the manager and search node(s)?

Thanks,
Wes

**Reply by mtoback** [reply]
No it's not! Thanks Wes! Is there a comprehensive list of ports anywhere?

**Reply by dougburks** [reply]
https://docs.securityonion.net/en/2.3/firewall.html#node-communication
